internal/db/schema/migrations/oss/postgres/28/01_prior_session_trigger.up.sql

```diff
@@ -0,0 +1,113 @@
+                select previous_end_time from session_state where session_id=target.session_id and state='terminated');
+            -- Then remove terminated end time
+            update session_state set end_time=NULL where session_id=target.session_id and state='terminated';
+        end if;
+    end loop;
+end; $$;
+
+-- Remove bad states for session ids with 3 states, similar to the above
+-- Difference from above is the need to check if terminated exists in the set of states
+-- Ex: PTA, PTC -> pruned to PT, PT
+-- Additional check for state PCA, pruned to PC
+-- Valid transitions like PAT, PCT, and in progress sessions like PAC will be ignored
+do $$
+declare
+    target record;
+begin
+    for target in select session_id from state_counts where count = 3 loop
+        -- If the last state is not terminated for this session...
+        if (select state from session_state where session_id in
+            (select session_id from state_counts where count = 3)
+            and session_id=target.session_id order by start_time desc limit 1)
+            != 'terminated' then
+            -- See if terminated appears; if so, prune back to it
+            if exists(select * from session_state where session_id=target.session_id and state='terminated')then
+                --Then we find the terminated record and timestamp and delete those that came before
+                delete from session_state where session_id = target.session_id and previous_end_time >(
+                    select previous_end_time from session_state where session_id=target.session_id and state='terminated');
+                -- Then remove terminated end time
+                update session_state set end_time=NULL where session_id=target.session_id and state='terminated';
+            end if;
+        end if;
+        -- Check for PCA case; if last state is not cancelled but canceling appears in the states
+        if (select state from session_state where session_id in
+            (select session_id from state_counts where count = 3)
+            and session_id=target.session_id order by start_time desc limit 1)
+            != 'canceling' then
+            -- See if canceling appears; if so, prune back to it
+            if exists(select * from session_state where session_id=target.session_id and state='canceling')then
+                --Then we find the canceling record and timestamp and delete those that came before
+                delete from session_state where session_id = target.session_id and previous_end_time >(
+                    select previous_end_time from session_state where session_id=target.session_id and state='canceling');
+                -- Then remove canceling end time
+                update session_state set end_time=NULL where session_id=target.session_id and state='canceling';
+            end if;
+        end if;
+    end loop;
+end; $$;
+
+-- Update insert session state transition trigger
```

Comment:
can you add a comment to reference the migration where this trigger was previously defined, ie:

```sql
-- Replaces trigger from 0/50_session.up.sql
```

And like-wise in `0/50_session.up.sql` can you add a comment to the original `create trigger` to note that it is replaced by this, ie:

```
-- Replaced in 28/01_prior_session_trigger.up.sql
````